File: gradle-plugins/cordform-common/src/main/java/net/corda/cordform/RpcSettings.java
package net.corda.cordform;

import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import com.typesafe.config.ConfigValueFactory;

public final class RpcSettings {

    private Config config = ConfigFactory.empty();

    private int port = 10003;
    private int adminPort = 10005;

    public int getPort() {
        return port;
    }

    public int getAdminPort() {
        return adminPort;
    }

    /**
     * RPC address for the node.
     */
    public final void address(final String value) {
        setValue("address", value);
    }

    /**
     * RPC Port for the node
     */
    public final void port(final int value) {
        this.port = value;
        setValue("address", "localhost:"+port);
    }

    /**
     * RPC admin address for the node (necessary if [useSsl] is false or unset).
     */
    public final void adminAddress(final String value) {
        setValue("adminAddress", value);
    }

    public final void adminPort(final int value) {
        this.adminPort = value;
        setValue("adminAddress", "localhost:"+adminPort);
    }

    /**
     * Specifies whether the node RPC layer will require SSL from clients.
     */
    public final void useSsl(final Boolean value) {
        setValue("useSsl", value);
    }

    /**
     * Specifies whether the RPC broker is separate from the node.
     */
    public final void standAloneBroker(final Boolean value) {
        setValue("standAloneBroker", value);
    }

    /**
     * Specifies SSL certificates options for the RPC layer.
     */
    public final void ssl(final SslOptions options) {
        config = options.addTo("ssl", config);
    }

    public final Config addTo(final String key, final Config config) {
        if (this.config.isEmpty()) {
            return config;
        }
        return config.withValue(key, this.config.root());
    }

    private void setValue(String path, Object value) {
        config = config.withValue(path, ConfigValueFactory.fromAnyRef(value));
    }
}

File: gradle-plugins/cordformation/src/main/kotlin/net/corda/plugins/RpcSettings.kt
package net.corda.plugins

import com.typesafe.config.Config
import com.typesafe.config.ConfigFactory
import com.typesafe.config.ConfigValueFactory
import groovy.lang.Closure
import org.gradle.api.Project

class RpcSettings(private val project: Project) {
    private var config: Config = ConfigFactory.empty()

    /**
     * RPC address for the node.
     */
    fun address(value: String) {
        config += "address" to value
    }

    /**
     * RPC admin address for the node (necessary if [useSsl] is false or unset).
     */
    fun adminAddress(value: String) {
        config += "adminAddress" to value
    }

    /**
     * Specifies whether the node RPC layer will require SSL from clients.
     */
    fun useSsl(value: Boolean) {
        config += "useSsl" to value
    }

    /**
     * Specifies whether the RPC broker is separate from the node.
     */
    fun standAloneBroker(value: Boolean) {
        config += "standAloneBroker" to value
    }

    /**
     * Specifies SSL certificates options for the RPC layer.
     */
    fun ssl(configureClosure: Closure<in SslOptions>) {
        val sslOptions = project.configure(SslOptions(), configureClosure) as SslOptions
        config = sslOptions.addTo("ssl", config)
    }

    internal fun addTo(key: String, config: Config): Config {
        if (this.config.isEmpty) {
            return config
        }
        return config + (key to this.config.root())
    }
}

internal operator fun Config.plus(entry: Pair<String, Any>): Config {

    return withValue(entry.first, ConfigValueFactory.fromAnyRef(entry.second))
}